presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/type/DecodedTimestamp.java

```diff
@@ -25,9 +25,7 @@
 
     public DecodedTimestamp(long epochSeconds, int nanosOfSecond)
     {
-        if (nanosOfSecond < 0 || nanosOfSecond >= NANOS_PER_SECOND) {
-            throw new IllegalArgumentException(format("Invalid value for nanosOfSecond: %s; must be positive and < %s", nanosOfSecond, NANOS_PER_SECOND));
-        }
+        checkArgument(nanosOfSecond >= 0 && nanosOfSecond < NANOS_PER_SECOND, "Invalid value for nanosOfSecond: " + nanosOfSecond);
```

Comment:
let the message be lazy fmt'd

```suggestion
        checkArgument(nanosOfSecond >= 0 && nanosOfSecond < NANOS_PER_SECOND, "Invalid value for nanosOfSecond: %s", nanosOfSecond);
```